### DISCUSSION TITLE
Single PR for a bundle of related KDE framework formulas

### DISCUSSION THREAD

**Discussion by wrobelda** [Homebrew/discussions#994] [Tap maintenance and Homebrew development] [answered]
Hi,

I understand that you prefer a single PR for each formula, but in case of KDE's KF5 framework formulas, it would make sense to bundle them together into a single PR. They are related semantically, logically and through their dependencies, while their formula definitions are mostly identical. 

Therefore I am asking if it is possible to have them sent in a single PR? 

We are talking about 30 (thirty) of the kf5-* formulas listed here (the second collapsed list): 
https://invent.kde.org/packaging/homebrew-kde/-/issues/23

**Comment by alebcay** [comment]
In such cases, we'll generally accept a PR that includes all of the changes together. However, we would still prefer one commit per formula within the PR (so the PR can contain multiple commits), as this keeps the history cleaner.

For submitting new formulae, I might still suggest limiting the number of formulae you submit at one time in a new PR. Troubleshooting and ironing out issues that arise with submitting new formulae only gets more complicated when there's multiple formulae involved (updating multiple formulae at once, on the other hand, is not as much of a hassle).

**Comment by carlocab** [comment, marked as answer]
You can probably submit them in small groups of 3-5 formulae at a time.

Or you can submit them all at once; it's up to you. I would advise against doing it all in one PR, however. That significantly increases the amount of work required in order to review the PR, which would not be to your advantage.

**Comment by yurikoles** [comment]
@wrobelda as I see from my experience every single formula requires some review iterations. Especially in case KF5, which require at least build-only test, even better to run it, if CI allows. You may see some examples in `kde-*` formulas I had submitted.

I started preparation of `kde-kconfig` [here](https://github.com/yurikoles/homebrew-core/blob/kde-kconfig/Formula/kde-kconfig.rb)

**Reply by wrobelda** [reply]
(deleted my previous response accidentally)

I also share your experience, but at the same time these formulas were already tested by many users, including ourselves, so I don't expect any teething problems here.  

I pushed my branch almost simultaneously here: https://github.com/wrobelda/homebrew-core/tree/extra_kf5_libs and created a PR for the first batch: https://github.com/Homebrew/homebrew-core/pull/72854
Let's continue the discussion there.

**Reply by yurikoles** [reply]
We use `ninja` in our tap because it is a bit faster than `make` and we have no bottles, but from `homebrew/core` POV it may be considered as avoidable extra dependency. E.g. invoking a 
```
brew create --cmake --set-name test-ecm https://download.kde.org/stable/frameworks/5.79/extra-cmake-modules-5.79.0.tar.xz
```

Produces the following boilerplate for install method:

```
def install
  # ENV.deparallelize  # if your formula fails when building in parallel
  system "cmake", ".", *std_cmake_args
  system "make", "install" # if this fails, try separate make/make install steps
end
```
